Src/RE2.Managed/Regex2.cs

```diff
@@ -186,6 +187,90 @@ public static void ClearRegexes()
             }
         }
 
+        public static unsafe void GetRegexSetup(string pattern, out ulong numCapturingGroups, out ulong numNamedCapturingGroups, out ulong numSubmatches, out ulong groupNameHeadersBufferSize, out ulong groupNamesBufferSize, out ulong submatchesBufferSize)
```

Comment:
>public [](start = 8, length = 6)

this could be internal, since the matches4 call it directly

Reply:
Yes, this probably won't exist after refactoring.

It should probably be integrated into the existing build-and-cache regex approach.

Reply:
This will probably be moved to `BuildRegex`.

Reply:
Refactoring `GetBuildRegex` moved to backlog: [https://github.com/microsoft/sarif-pattern-matcher/issues/386](https://github.com/microsoft/sarif-pattern-matcher/issues/386)

Reply:
For now, I've made it private.